### ISSUE TITLE
Feature Request: Add QR Code to Downloaded PDFs

### ISSUE THREAD

**Issue by knobunc** [eikek/docspell#2013]
As a user of docspell, when I print a doc, it would be amazing to be able to (optionally) add a small QR code to all of the pages so that I could easily tie the printed copy back to the stored file.

**Comment by knobunc** [comment]
I get a bunch of pdfs through email (or that I have scanned).  But when I later print them, it would be really handy to be able to scan a QR code on the print to get back to the doc in docspell.

Specifically, I am building a house and we get lots of design docs, and each has lots of versions.  So if I printed one to give to a contractor, when we are talking about it, it would be awesome to be able to scan a link to get to docspell, and then I can look at the linked documents to see if there is a later revision we should really be looking at, or if this one is current.

Obviously, it should be an option to turn on/off at download time... if it were always on for converted pdfs it would be annoying.

**Comment by madduck** [comment]
It should be possible to just overlay these data on the fly when generating the PDF for download, that doesn't require processing the entire PDF I don't think.

**Comment by eikek** [comment]
> It should be possible to just overlay these data on the fly when generating the PDF for download, that doesn't require processing the entire PDF I don't think.

That's true. I'm only not sure if the pdf library in use wants to have the complete PDF in memory to do any operation. I think in the latest release there were changes in that respect, but I have yet to really take a look.

**Comment by madduck** [comment]
Is this PDFBox? If so, I can have a look…

**Comment by eikek** [comment]
@stefan123t attachments and items have different ids. There are two ways for obtaining a qr code: one is for the item - it is in the menu below the item header. The other is in the menu next to the attachment.

**Comment by stefan123t** [comment]
@eikek thanks for explaining the difference of the two QR codes! I found it in the menu of the Item (see scrrenshot below).

The Tooltip reads **Show URL as QR Code**, suggesting to change the title to **Show Item URL as QR Code** vs. **Show Attachement URL as QR Code** to make the two distinguishable also for users requiring aria accessibility annotations.

<img width="730" height="588" alt="Image" src="https://github.com/user-attachments/assets/7cd0de42-99b3-45de-8b93-77d3cf76fb05" />
